### PR TITLE
Add ability to receive Unknown events

### PR DESCRIPTION
**Pull request by DRSchlaubi**
/cc @gdude2002

### CODE REVIEW

**Review by gdude2002 — APPROVED** [review]
LGTM. Appreciate this.

File: core/src/main/kotlin/event/Event.kt [hunk 1]
```diff
@@ -45,3 +51,17 @@ public interface Event : KordObject {
...
     public val customContext: Any?
 }
+
+/**
+ * Representation of an event that has not been documented / added to Kord yet.
+ *
+ * @property data the raw [json body][JsonElement] of this event if provided
+ * @property name the name of the event if provided
+ */
+public data class UnknownEvent(
```
Comment (Lukellmann) [inline core/src/main/kotlin/event/Event.kt:61]: other events aren't data classes, `equals`, `hashCode` and `copy` don't make much sense for events
i think we should instead just provide a custom `toString` implementation

File: core/src/main/kotlin/gateway/handler/DefaultGatewayEventInterceptor.kt [hunk 2]
```diff
@@ -36,6 +36,7 @@ public class DefaultGatewayEventInterceptor @KordPreview public constructor(
     public constructor() : this(customContextCreator = null)
 
     private val listeners = listOf(
+        UnknownEventHandler(),
```
Comment (Lukellmann) [inline core/src/main/kotlin/gateway/handler/DefaultGatewayEventInterceptor.kt:39]: i ordered the other handlers alphabetically to have the same order as the files in the ide, could you put it in the right place?

File: core/src/main/kotlin/event/Event.kt [hunk 3]
```diff
@@ -45,3 +51,17 @@ public interface Event : KordObject {
...
+    override val kord: Kord,
+    override val shard: Int,
+    override val customContext: Any?,
+    val data: JsonElement?,
+    val name: String?
```
Comment (Lukellmann) [inline core/src/main/kotlin/event/Event.kt:66]: i think `name` should be the first and `data` the second parameter an they should be put before `kord`, `shard` and `customContext` to match other events

File: gateway/src/main/kotlin/Event.kt [hunk 4]
```diff
@@ -805,6 +805,12 @@ public data class GuildScheduledEventUserRemove(
     override val sequence: Int?,
 ) : DispatchEvent()
 
+public data class UnknownDispatchEvent(
+    val data: JsonElement?,
+    override val sequence: Int?,
+    val name: String?
```
Comment (Lukellmann) [inline gateway/src/main/kotlin/Event.kt:811]: i think `name` should be the first parameter

**Review by Lukellmann — COMMENTED** [review]
only some minor things that i think should be adressed but looks good overall

File: core/src/main/kotlin/gateway/handler/DefaultGatewayEventInterceptor.kt (hunk 2 above)
Comment (DRSchlaubi) [reply core/src/main/kotlin/gateway/handler/DefaultGatewayEventInterceptor.kt:39]: sure

File: core/src/main/kotlin/event/Event.kt [hunk 5]
```diff
@@ -45,3 +51,21 @@ public interface Event : KordObject {
...
+public class UnknownEvent(
+    public val name: String?,
+    public val data: JsonElement?,
+    override val kord: Kord,
+    override val shard: Int,
+    override val customContext: Any?,
+) : Event {
+    override fun toString(): String {
+        return "UnknownEvent(kord=$kord, shard=$shard, customContext=$customContext, data=$data, name=$name)"
+    }
```
Comment (Lukellmann) [inline core/src/main/kotlin/event/Event.kt:70]:
```suggestion
    override fun toString(): String =
        "UnknownEvent(name=$name, data=$data, kord=$kord, shard=$shard, customContext=$customContext)"
```
to match the order of the constructor parameters

File: gateway/src/main/kotlin/Event.kt [hunk 6]
```diff
@@ -491,6 +535,7 @@ public sealed class Event {
                     ),
                     sequence
                 )
+
```
Comment (Lukellmann) [inline gateway/src/main/kotlin/Event.kt:538]: could you remove the line breaks again? i think `)` having their own lines is enough spacing between branches (maybe adjust your formatter)

**Review by Lukellmann — COMMENTED** [review]
almost approved :)

**Review by Lukellmann — APPROVED** [review]
looks good, just gonna write up a commit message and then merge
thanks :)

**Pull request merged**